src/Pate/ExprMappable.hs

```diff
@@ -132,3 +140,21 @@ instance ExprMappable sym (SkipTransformation a) where
 
 instance (Ord f, ExprMappable sym f) => ExprMappable sym (WPM.PredMap sym f k) where
   mapExpr sym f pm = WPM.alter sym pm (\v p -> (,) <$> mapExpr sym f v <*> f p)
+
+instance (OrdF f, ExprMappable sym (f tp)) => ExprMappable sym (SetF.SetF f tp) where
+  mapExpr sym f s = SetF.fromList <$> traverse (mapExpr sym f) (SetF.toList s)
+
+newtype SymExprMappable sym f = SymExprMappable (forall tp a. ((ExprMappable sym (f tp)) => a) -> a)
+
+-- | Deduce ad-hoc 'ExprMappable' instances for 'WI.SymExpr'.
+--   This uses an unsafe coercion to implicitly use the 'ExprMappable' instance
+--   defined in 'ToExprMappable'.
+--   This is necessary because 'WI.SymExpr' is a type family and therefore we cannot
+--   declare the obvious instance for it.
+symExprMappable ::
+  forall sym.
+  sym ->
+  SymExprMappable sym (WI.SymExpr sym)
+symExprMappable _sym = unsafeCoerce r
```

Comment:
This is quite scary.

Why is this safe? Is there no other way to achieve what you need here?